### PR TITLE
Chore: Add end-to-end tests to teams listing in the `directory` endpoint

### PR DESCRIPTION
**Pull request by carlosrodrigues94**

### CODE REVIEW

**Review by matheusbsilva137 — DISMISSED** [review]

**Comment by sampaiodiego** [comment]
@carlosrodrigues94 did you check the CI result? it seems to be failing in the new test you just created =)

**Comment by carlosrodrigues94** [comment]
> @carlosrodrigues94 did you check the CI result? it seems to be failing in the new test you just created =)

fixed.

**Review by matheusbsilva137 — APPROVED** [review]

File: apps/meteor/tests/end-to-end/api/00-miscellaneous.js [hunk 1]
```diff
@@ -389,6 +389,59 @@ describe('miscellaneous', function () {
...
+					sort: JSON.stringify({
+						name: 1,
+					}),
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('result');
+					expect(res.body.result).to.be.an(`array`);
+					expect(res.body).to.have.property('total');
```
Comment (sampaiodiego) [inline apps/meteor/tests/end-to-end/api/00-miscellaneous.js:430]: ideally, you should also check if the values returned by the endpoint are the expected ones.. for example, if you have created just a single team but it returns `total: 2`, it probably means it is returning not only teams, so there is probably a bug on the endpoint.
```suggestion
					expect(res.body).to.have.property('total', 1);
```

File: apps/meteor/tests/end-to-end/api/00-miscellaneous.js [hunk 2]
```diff
@@ -389,6 +389,59 @@ describe('miscellaneous', function () {
...
+					expect(res.body.total).to.be.an('number');
+					expect(res.body.result[0]).to.have.property('_id');
```
Comment (sampaiodiego) [inline apps/meteor/tests/end-to-end/api/00-miscellaneous.js:432]:
```suggestion
					expect(res.body.result[0]).to.have.property('_id', teamIdCreatedPreviously);
```

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Comment by carlosrodrigues94** [comment]
> # [Codecov](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat) Report
> 
> > Merging [#26347](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat) ([94b805d](https://github.com/RocketChat/Rocket.Chat/commit/94b805d3b1568dc386bc90272601e212220fe771)) into [develop](https://codecov.io/gh/RocketChat/Rocket.Chat/commit/355819e6fbeba67826db4fe1c2bfc26d24db0d61?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat) ([355819e](https://github.com/RocketChat/Rocket.Chat/commit/355819e6fbeba67826db4fe1c2bfc26d24db0d61)) will **decrease** coverage by `0.00%`.
> > The diff coverage is `n/a`.
> 
> ```diff
> @@             Coverage Diff             @@
> ##           develop   #26347      +/-   ##
> ===========================================
> - Coverage    39.05%   39.05%   -0.01%     
> ===========================================
>   Files          748      748              
>   Lines        18647    18647              
>   Branches      1424     1424              
> ===========================================
> - Hits          7283     7282       -1     
> - Misses       11141    11143       +2     
> + Partials       223      222       -1     
> ```
> 
> Flag 	Coverage Δ 	
> e2e 	`39.05% <ø> (-0.01%)` 	⬇️
> 
> Flags with carried forward coverage won't be shown. [Click here](https://docs.codecov.io/docs/carryforward-flags?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat#carryforward-flags-in-the-pull-request-comment) to find out more.
> [Impacted Files](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat) 	Coverage Δ 	
> [...app/ui-message/client/popup/messagePopupChannel.js](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat#diff-YXBwcy9tZXRlb3IvYXBwL3VpLW1lc3NhZ2UvY2xpZW50L3BvcHVwL21lc3NhZ2VQb3B1cENoYW5uZWwuanM=) 	`50.00% <0.00%> (-50.00%)` 	⬇️
> [...views/room/MessageList/lib/isMessageFirstUnread.ts](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat#diff-YXBwcy9tZXRlb3IvY2xpZW50L3ZpZXdzL3Jvb20vTWVzc2FnZUxpc3QvbGliL2lzTWVzc2FnZUZpcnN0VW5yZWFkLnRz) 	`50.00% <0.00%> (-16.67%)` 	⬇️
> [apps/meteor/client/lib/meteorCallWrapper.ts](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat#diff-YXBwcy9tZXRlb3IvY2xpZW50L2xpYi9tZXRlb3JDYWxsV3JhcHBlci50cw==) 	`96.42% <0.00%> (-3.58%)` 	⬇️
> [apps/meteor/app/ui-login/client/login/form.js](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat#diff-YXBwcy9tZXRlb3IvYXBwL3VpLWxvZ2luL2NsaWVudC9sb2dpbi9mb3JtLmpz) 	`73.20% <0.00%> (-0.66%)` 	⬇️
> [...ached-collection/client/models/CachedCollection.js](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat#diff-YXBwcy9tZXRlb3IvYXBwL3VpLWNhY2hlZC1jb2xsZWN0aW9uL2NsaWVudC9tb2RlbHMvQ2FjaGVkQ29sbGVjdGlvbi5qcw==) 	`85.41% <0.00%> (-0.53%)` 	⬇️
> [...teor/app/ui-utils/client/lib/RoomHistoryManager.js](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat#diff-YXBwcy9tZXRlb3IvYXBwL3VpLXV0aWxzL2NsaWVudC9saWIvUm9vbUhpc3RvcnlNYW5hZ2VyLmpz) 	`51.44% <0.00%> (-0.49%)` 	⬇️
> [apps/meteor/app/ui/client/views/app/room.js](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat#diff-YXBwcy9tZXRlb3IvYXBwL3VpL2NsaWVudC92aWV3cy9hcHAvcm9vbS5qcw==) 	`49.59% <0.00%> (-0.21%)` 	⬇️
> [apps/meteor/app/ui-utils/client/lib/RoomManager.js](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat#diff-YXBwcy9tZXRlb3IvYXBwL3VpLXV0aWxzL2NsaWVudC9saWIvUm9vbU1hbmFnZXIuanM=) 	`76.00% <0.00%> (+1.14%)` 	⬆️
> [...r/client/startup/notifications/usersNameChanged.ts](https://codecov.io/gh/RocketChat/Rocket.Chat/pull/26347/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat#diff-YXBwcy9tZXRlb3IvY2xpZW50L3N0YXJ0dXAvbm90aWZpY2F0aW9ucy91c2Vyc05hbWVDaGFuZ2VkLnRz) 	`100.00% <0.00%> (+66.66%)` 	⬆️
> 
> Help us with your feedback. Take ten seconds to tell us [how you rate us](https://about.codecov.io/nps?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=RocketChat).

updating asserts to ensure coverage

**Pull request merged**